.ci/release.groovy

```diff
@@ -0,0 +1,124 @@
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    timeout(time: 2, unit: 'HOURS')
+    disableConcurrentBuilds()
+  }
+  parameters {
+    choice(choices: ['snapshot', 'staging', 'production'], description: 'Branch and environment to Release the distribution.', name: 'environment')
+    string(name: 'commit', defaultValue: "", description: "Commit SHA1/Branch tag of the Docker image.")
+  }
+  stages {
+    stage('Select Docker Image'){
+      when {
+        expression { return '' == params.commit }
+      }
+      steps {
+        gitCheckout(basedir: 'src', branch: "${env.BRANCH_NAME}",
+          repo: "git@github.com:elastic/${REPO}.git",
+          credentialsId: "${env.JOB_GIT_CREDENTIALS}"
+        )
+        dir('src'){
+          sh(label: 'Get latest commits', script: '''
+            echo "[" > latest-changes.json
+            git log -n 5 --pretty=format:'{"sha":"%H","msg":"%s","author":"%an"},' >> latest-changes.json
+            echo "]" >> latest-changes.json
+          ''')
+          script {
+            def json = readJSON(file: 'latest-changes.json')
+            def options = json?.collect{item -> "${item.msg}-${item.sha}"}
+            def choose = input(message: "Choose the version to release to ${env.BRANCH_NAME}",
+                        parameters: [
+                          choice(choices: options, 
+                            description: 'Select the commit to release',
+                            name: 'tag')
+                        ]
+                      )
+            setEnvVar('TAG', choose[-40..-1])
+            echo 
+          }
+        }
+      }
+    }
+    stage('Release') {
+      steps {
+        deleteDir()
+        gitCheckout(basedir: 'infra', branch: "master",
+          repo: "git@github.com:elastic/infra.git",
+          credentialsId: "${env.JOB_GIT_CREDENTIALS}")
+        dir("infra"){
+          sshagent(["${env.JOB_GIT_CREDENTIALS}"]) {
+            sh(label: 'Create local branch', script: """
+              git checkout -b ${REPO}-release
+            """)
+            script {
+              def ymlFile = "terraform/providers/gcp/env/elastic-apps-web/helm/values/package-registry/package-registry-${env.BRANCH_NAME}.yaml"
+              def yml = readYaml(file: ymlFile)
+              yml.image.tag = "${env.TAG}"
+              writeYaml(file: ymlFile, data: yml, overwrite: true)
+              setEnvVar('DOCKER_IMAGE', yml.image.repository)
+            }
+            sh(label: 'Git config', script: """
+              git config --global user.name observability-robots
+              git config --global user.email observability-robots@users.noreply.github.com
+            """)
+            sh(label: 'Commit local changes', script: "git commit -a -m '[package-registry]: Release ${env.TAG} to ${env.BRANCH_NAME}'")
+            sh(label: 'Push changes', script: "git push origin package-registry-${env.REPO}-release --force")
```

Comment:
Could it be replaced with the below steps?
- `withGitRelease()` 
-  `gitPush()`